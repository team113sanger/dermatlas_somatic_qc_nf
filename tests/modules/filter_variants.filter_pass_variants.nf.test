nextflow_process {

    name "Test Process FILTER_PASS_VARIANTS"
    script "modules/filter_variants.nf"
    process "FILTER_PASS_VARIANTS"

    test("Should run without failures") {

        when {
        params {
            load("$baseDir/tests/testdata/test_params.json")
        }
            process {
                """
                input[0] = tuple([sample_id: "PD54555a", caller: "caveman"], 
                                 file("/lustre/scratch124/jb63/mafqc/caveman_files/PD54555a/PD54555a.smartphase.vep.vcf.gz"))
                input[1] = file("${params.baitset}")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
