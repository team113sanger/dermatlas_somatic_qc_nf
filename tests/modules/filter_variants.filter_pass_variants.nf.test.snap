{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sample_id": "PD54555a",
                            "caller": [
                                [
                                    "PD54555a.filt.vcf.gz:md5,712d17002b923025ea3b8217536f6079",
                                    "PD54555a.filt.vcf.gz.tbi:md5,ae87bd4847e2ee6028288e7f5ffd53ac"
                                ],
                                [
                                    "PD54556a.filt.vcf.gz:md5,5668282c302bceaef63a5a74950b8ee8",
                                    "PD54556a.filt.vcf.gz.tbi:md5,072ac6c7801c9204cebab8a4a26e1f46"
                                ],
                                [
                                    "PD54557a.filt.vcf.gz:md5,1140309e6ad9052d272371fd2a6ec922",
                                    "PD54557a.filt.vcf.gz.tbi:md5,84678be53e42369799dcd7522ff11833"
                                ],
                                [
                                    "PD54558a.filt.vcf.gz:md5,a0387b3e07e6678873244588d140a879",
                                    "PD54558a.filt.vcf.gz.tbi:md5,bd1e40317dfc95b23cdef67635d762b1"
                                ]
                            ]
                        },
                        "PD54555a.filt.vcf.gz:md5,dc8077b123c8e7dc3948631f6d2cf490",
                        "PD54555a.filt.vcf.gz.tbi:md5,a224e16fc048581de986b74f7cd1cdeb"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-18T11:59:34.184893"
    }
}